tests/lib/eslint.js

```diff
@@ -3189,6 +3189,32 @@ describe("eslint", function() {
             eslint.verify("var { a='a' } = {};", { parserOptions: { ecmaVersion: 6 }});
         });
 
+        it("should not rewrite env setting in core (https://github.com/eslint/eslint/issues/4814)", function() {
+            eslint.defineRule("test", function() {
+                return {};
+            });
+            eslint.verify("var a = 0;", {
+                env: {node: true},
```

Comment:
This test focuses the instance of [`{globalReturn: true}`](https://github.com/eslint/eslint/blob/v2.0.0-alpha-2/conf/environments.js#L26-L28)

The first `verify()` modifies the instance with `{globalReturn: false}` at https://github.com/eslint/eslint/blob/v2.0.0-alpha-2/lib/eslint.js#L416, before this PR.
Then the second `verify()` takes and checks the instance was not modified.

#4814 is doing an additional thing, `RuleTester` freezes the instance.
Then the second `verify()` throws an error at https://github.com/eslint/eslint/blob/v2.0.0-alpha-2/lib/eslint.js#L416 since the instance was frozen.

After this PR, when `verify()` takes environment object, it copies the instance deeply, so the second `verify()` works as expected.